### PR TITLE
[ROCm] Enable ROCm support for "softplus_op"

### PR DESCRIPTION
**Pull request by deven-amd**
This PR enables ROCm support for the "softplus_op".

PR #25676  is a pre-requisite for this PR. Since that is not yet merged the commits for that PR are included in this one as well (the first 6 commits), but should be ignored for review purposes.

Of the 4 commits exclusive to this PR, the first 3 are for minor fixes required for dynamic loading of ROCm runtime libraries. Only the last commit pertains to the "softplus_op".

@tatianashp : just FYI
@whchung

### CODE REVIEW

**Comment by jlebar** [comment]
@rthadur, hey, as in

  https://github.com/tensorflow/tensorflow/pull/25796#issuecomment-465241088
  https://github.com/tensorflow/tensorflow/pull/25676#issuecomment-462574555
  https://github.com/tensorflow/tensorflow/pull/26342#issuecomment-469904659

because this does not touch XLA, I am not the right person to review this code.

Is there a doc somewhere that I can update that will help you find the right person more easily for these changes?

**Comment by rthadur** [comment]
> @rthadur, hey, as in
> 
> [#25796 (comment)](https://github.com/tensorflow/tensorflow/pull/25796#issuecomment-465241088)
> [#25676 (comment)](https://github.com/tensorflow/tensorflow/pull/25676#issuecomment-462574555)
> [#26342 (comment)](https://github.com/tensorflow/tensorflow/pull/26342#issuecomment-469904659)
> 
> because this does not touch XLA, I am not the right person to review this code.
> 
> Is there a doc somewhere that I can update that will help you find the right person more easily for these changes?

@jlebar will make a note of it and make sure will assign correct person.

**Comment by chsigg** [comment]
Hi Deven, PR #25676 has landed (finally, thanks for your patience). Would you mind to sync this one to HEAD? I can start reviewing it early next week. Thank you!

**Comment by deven-amd** [comment]
@chsigg 

rebasing done....much smaller PR now.   Also most of the changes are "internal" to ROCm. 

Thanks again for getting the ROCm PRs reviewed and merged.

deven

**Review by chsigg — APPROVED** [review]
This PR contains a little more than just enabling softplus_op. That's ok, but maybe update the PR title.

Other than that, it looks good.

File: tensorflow/stream_executor/rocm/BUILD [hunk 1]
```diff
@@ -232,7 +232,6 @@ cc_library(
         "//tensorflow/core:lib_internal",
         "//tensorflow/stream_executor:dnn",
         "//tensorflow/stream_executor:event",
-        "//tensorflow/stream_executor:logging_proto_cc",
```
[the remark this answers is not in the document]
Comment (deven-amd) [reply tensorflow/stream_executor/rocm/BUILD]: very much so.

Leaving it in is what breaks the `--config=rocm` build.

The CUDA equivalent was removed in the following commit : 
https://github.com/tensorflow/tensorflow/commit/5aefc4e922694df4ebec80784dcd962ac2d31a5f

**Comment by deven-amd** [comment]
@chsigg 

Thank you for getting this merged.

The merge process seems to have picked up the wrong line for one the commits.

change in this PR : https://github.com/tensorflow/tensorflow/pull/26457/files?file-filters%5B%5D=.bzl#diff-38c70d53f71218cab252a7f458600d6dR1212

change that actually got commited : https://github.com/tensorflow/tensorflow/commit/500d747dadaa6eb3cf11f9190f741a500a751ce4#diff-38c70d53f71218cab252a7f458600d6dR1237

It seems like the merge was off by one line.

I will fix this when I rebase the PR for `softsign` op.

**Pull request merged**